### PR TITLE
add v1 to various client schemes

### PR DESCRIPTION
**Pull request by dprotaso**
**What type of PR is this?**
/kind cleanup
/kind test
/area conformance

**What this PR does / why we need it**:

Gateway API v1 APIs need to be added the scheme. Discovered in https://github.com/kubernetes-sigs/gateway-api/pull/2456#pullrequestreview-1690743992 - thanks @sunjayBhatia 

**Which issue(s) this PR fixes**:
N/A

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Review by sunjayBhatia — APPROVED** [review]

File: conformance/conformance_test.go [hunk 1]
```diff
@@ -20,6 +20,7 @@ package conformance_test
 import (
 	"testing"
 
+	v1 "sigs.k8s.io/gateway-api/apis/v1"
```
Comment (sunjayBhatia) [inline conformance/conformance_test.go:23]: ha maybe don't need the alias for the files that don't already

maybe we should have a lint rule to make them consistent, but that can be done separately

Comment (dprotaso) [reply conformance/conformance_test.go:23]: >ha maybe don't need the alias for the files that don't already

goimports forces it - i'm gonna assume verify will pass. If we want to tweak it I'd say let's do this separate from this PR

**Review by sunjayBhatia — APPROVED** [review]

**Comment by robscott** [comment]
Thanks @dprotaso!

/lgtm
/approve

**Pull request merged**